core/src/main/java/org/elasticsearch/ElasticsearchCorruptionException.java

```diff
@@ -42,10 +42,7 @@ public ElasticsearchCorruptionException(String message) {
      * @param ex the exception cause
      */
     public ElasticsearchCorruptionException(Throwable ex) {
-        this(ex.getMessage());
-        if (ex != null) {
-            this.setStackTrace(ex.getStackTrace());
-        }
+        super(ex);
         Throwable[] suppressed = ex.getSuppressed();
```

Comment:
Well, it shouldnt be. The suppressed exceptions are like 'children' of `cause`, they don't need to be shuffled around. The current code is basically attaching grandchildren as children, which will be confusing.